dallinger/command_line/__init__.py

```diff
@@ -523,49 +508,63 @@ def _current_hits(service, app):
     flag_value=True,
     help="Look for HITs in the MTurk sandbox rather than the live/production environment",
 )
-def hits(app, sandbox):
-    """List all HITs for the user's configured MTurk request account,
-    or for a specific experiment id.
-    """
+@click.option("--recruiter", default="mturk", help="Experiment id")
+def hits(app, sandbox, recruiter):
+    """List all HITs for the recruiter account or for a specific experiment id."""
     if app is not None:
         verify_id(None, "--app", app)
-    formatted_hit_list = []
-    dateformat = "%Y/%-m/%-d %I:%M:%S %p"
-    for h in _current_hits(_mturk_service_from_config(sandbox), app):
-        title = h["title"][:40] + "..." if len(h["title"]) > 40 else h["title"]
-        description = (
-            h["description"][:60] + "..."
-            if len(h["description"]) > 60
-            else h["description"]
-        )
-        formatted_hit_list.append(
-            [
-                h["id"],
-                title,
-                h["annotation"],
-                h["status"],
-                h["created"].strftime(dateformat),
-                h["expiration"].strftime(dateformat),
-                description,
-            ]
-        )
-    out = Output()
-    out.log("Found {} hit[s]:".format(len(formatted_hit_list)))
-    out.log(
-        tabulate.tabulate(
-            formatted_hit_list,
-            headers=[
-                "Hit ID",
-                "Title",
-                "Annotation (experiment ID)",
-                "Status",
-                "Created",
-                "Expiration",
-                "Description",
-            ],
-        ),
-        chevrons=False,
-    )
+
+    rec = by_name(recruiter, not_validate_config=True)
+    rec.hits(app, sandbox)
+
+
+@dallinger.command()
+@click.option("--hit_id", default=None, help="MTurk HIT ID")
+@click.option(
+    "--sandbox",
+    is_flag=True,
+    flag_value=True,
+    help="Look for HITs in the MTurk sandbox rather than the live/production environment",
+)
+@click.option("--recruiter", default="mturk", help="Experiment id")
+def hit_details(hit_id, sandbox, recruiter):
+    """Print the details for a specific HIT is for a recruiter."""
+    rec = by_name(recruiter, not_validate_config=True)
+    details = rec.hit_details(hit_id, sandbox)
+    print(json.dumps(details, indent=4, default=str))
+
+
+@dallinger.command()
+@click.option("--hit_id", default=None, help="MTurk HIT ID")
+@click.option(
+    "--sandbox",
+    is_flag=True,
+    flag_value=True,
+    help="Look for HITs in the MTurk sandbox rather than the live/production environment",
+)
+@click.option("--recruiter", default="mturk", help="Experiment id")
+@click.option(
+    "--qualification_path",
+    default=None,
+    help="Filename/path for the qualification file",
+)
+def copy_qualifications(hit_id, sandbox, recruiter, qualification_path):
+    """Copy qualifications from an existing HIT ID."""
```

Comment:
Extend documentation to something like 'Copy qualifications from an existing HIT and save them to a JSON file`?

Reply:
Done